### PR TITLE
Implementing a GORM serializer for `timestamppb.Timestamp`

### PR DESCRIPTION
**Pull request by oxisto**
This allows to serialize protobuf messages that contain timestamps into our GORM database.

### CODE REVIEW

File: persistence/gorm/gorm.go [hunk 1]
```diff
@@ -206,3 +213,45 @@ func (s *storage) Delete(r any, conds ...any) error {
 
 	return nil
 }
+
+// TimestampSerializer is a GORM serializer that allows the serialization and unserialization of the
+// google.protobuf.Timestamp protobuf message type.
```
Comment (lebogg) [inline persistence/gorm/gorm.go:218]:
```suggestion
// TimestampSerializer is a GORM serializer that allows the serialization and deserialization of the
```

File: persistence/gorm/gorm_test.go [hunk 2]
```diff
@@ -466,3 +485,98 @@ func Test_storage_Delete(t *testing.T) {
...
+		},
+		{
+			name: "field wrong type",
+			args: args{
+				field:      &schema.Field{Name: "timestamp"},
+				dst:        reflect.Value{},
+				fieldValue: "string",
+			},
+			want:    nil,
+			wantErr: true,
```
Comment (lebogg) [inline persistence/gorm/gorm_test.go:531]: Test directly against "ErrUnsupportedType"

File: service/discovery/discovery.go [hunk 3]
```diff
@@ -53,6 +52,7 @@ import (
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 	"google.golang.org/protobuf/types/known/structpb"
+	"google.golang.org/protobuf/types/known/timestamppb"
```
Comment (lebogg) [inline service/discovery/discovery.go:55]: Made the imports a lil bit better but not complete right ^^

**Review by lebogg — COMMENTED** [review]
LGMT. Just some minor things.

File: api/assessment/assessment.proto [hunk 4]
```diff
@@ -102,7 +103,8 @@ message AssessmentResult {
   string id = 1;
 
   // Time of assessment
-  google.protobuf.Timestamp timestamp = 2;
+  google.protobuf.Timestamp timestamp = 2
+      [ (tagger.tags) = "gorm:\"serializer:timestamppb;type:time\"" ];
```
[the remark this answers is not in the document]
Comment (oxisto) [reply api/assessment/assessment.proto:107]: I have not seen that. Where?

[the remark this answers is not in the document]
Comment (lebogg) [reply api/assessment/assessment.proto:107]: For example in the [manytomany](https://gorm.io/docs/many_to_many.html) example:
```
Languages []Language `gorm:"many2many:user_languages;"`
```

[the remark this answers is not in the document]
Comment (oxisto) [reply api/assessment/assessment.proto:107]: My guess that this is a mistake, in all other examples, there is no trailing `;`

**Pull request merged**